app/tests/test_collections_endpoint.py

```diff
@@ -84,3 +86,130 @@ def test_collections_limit_query(self):
+        )
+        response_json = response.json()
+        self.assertEqual(response.status_code, 400, msg=get_http_error_description(response_json))
+
+    def test_collections_min_mandatory_post(self):
+        # a post with the absolute valid minimum
+        payload_json = self.sample_collections['valid_min_collection_set_1']
+
+        response = self.client.post(
+            f"/{API_BASE}/collections", data=payload_json, content_type='application/json'
+        )
+        response_json = response.json()
+        logger.debug(response_json)
+        self.assertEqual(response.status_code, 201, msg=get_http_error_description(response_json))
+        self.assertNotIn('title', response_json.keys())  # key does not exist
+        self.assertEqual(response_json['providers'], [])
+
+    def test_collections_less_than_mandatory_post(self):
+        # a post with the absolute valid minimum
+        payload_json = self.sample_collections['less_than_min_collection_set']
+
+        response = self.client.post(
+            f"/{API_BASE}/collections", data=payload_json, content_type='application/json'
+        )
+        response_json = response.json()
+        self.assertEqual(response.status_code, 400, msg=get_http_error_description(response_json))
+
+    def test_collections_put(self):
+        payload_json = self.sample_collections['valid_collection_set_2']
+
+        # the dataset to update does not exist yet
+        response = self.client.put(
+            f"/{API_BASE}/collections/{payload_json['id']}",
+            data=payload_json,
+            content_type='application/json'
+        )
+        response_json = response.json()
+        self.assertEqual(response.status_code, 404, msg=get_http_error_description(response_json))
+
+        # POST data
+        self.client.post(
+            f"/{API_BASE}/collections", data=payload_json, content_type='application/json'
+        )
+
+        payload_json['title'] = "The Swallows"
+        response = self.client.put(
+            f"/{API_BASE}/collections/{payload_json['id']}",
+            data=payload_json,
+            content_type='application/json'
+        )
+        response_json = response.json()
+
+        self.assertEqual(response.status_code, 200, msg=get_http_error_description(response_json))
+        self.assertEqual(response_json['title'], payload_json['title'])
+
+        # is it persistent?
+        response = self.client.get(
+            f"/{API_BASE}/collections/{payload_json['id']}",
+            data=payload_json,
+            content_type='application/json'
+        )
+
+        response_json = response.json()
+
+        self.assertEqual(response.status_code, 200, msg=get_http_error_description(response_json))
+        self.assertEqual(response_json['title'], payload_json['title'])
+
+    def test_collection_put_change_id(self):
+        payload_json = self.sample_collections['valid_collection_set_3']
+        logger.debug(self.collections[0].name)
```

Comment:
Maybe an assert to be sure that `collection.id`  in `self.collections[0]` and in `payload_json` would be good here. To avoid changing names in future and breaking this unittest without knowing it.